airbyte-integrations/connectors/source-iterable/source_iterable/api.py

```diff
@@ -0,0 +1,231 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+import urllib.parse as urlparse
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from base_python import HttpStream
+
+
+class IterableStream(HttpStream, ABC):
+    url_base = "https://api.iterable.com/api/"
+
+    # Hardcode the value because it is not returned from the API
+    BACKOFF_TIME_CONSTANT = 10.0
```

Comment:
I saw the default behavior for backing off and find that we can't use the default backoff for 429 error, because we just ignore it.
https://github.com/airbytehq/airbyte/blob/3e6843808e484a99e53843458c308a195779ebf7/airbyte-integrations/bases/base-python/base_python/sdk/streams/rate_limiting.py#L45-L47

Reply:
Ah I see. I have a fix for this in the latest refactor of the framework -- this is a bug. It should have `and  exc.response.status_code != 429`. We'll be able to refactor this soon to rely on the default backoff behavior.